node/network/gossip-support/src/lib.rs

```diff
@@ -277,21 +275,31 @@ where
 		Ok(())
 	}
 
-	async fn update_authority_status_metrics(&mut self, session_info: &SessionInfo) {
-		let maybe_index =
-			match ensure_i_am_an_authority(&self.keystore, &session_info.discovery_keys).await {
-				Ok(index) => {
-					self.metrics.on_is_authority();
-					Some(index)
-				},
-				Err(util::Error::NotAValidator) => {
-					self.metrics.on_is_not_authority();
-					self.metrics.on_is_not_parachain_validator();
-					None
-				},
-				// Don't update on runtime errors.
-				Err(_) => None,
-			};
+	// Checks if the node is an authority and also updates `polkadot_node_is_authority` and
+	// `polkadot_node_is_parachain_validator` metrics accordingly.
+	// On success, returns the index of our keys in `session_info.discovery_keys`.
+	async fn check_authority_status_and_update_metrics(
```

Comment:
Maybe `get_authority_index_and_update_metrics`? `check` is a bit misleading here.